src/gateway/http.ts

```diff
@@ -61,9 +62,9 @@ export class HTTP extends Gateway {
 
     const handler = defaults.protocol === 'https:' ? https : http
 
-    const requestParams: HTTPRequestParams = assign(defaults, {
-      method: requestMethod,
-      headers: assign(headers, this.request.headers()),
+    const requestParams: http.RequestOptions = assign(defaults, {
+      method,
+      headers: assign(headers, this.request.headers() as http.OutgoingHttpHeaders),
```

Comment:
Why do we need the cast now?

Reply:
Because I added `const requestParams: http.RequestOptions` to make it easier to reason about this object, such as being able to do `requestParams.signal = ...` and getting type safety. However this clashes a bit with how `this.request.headers()` is typed:

```
Type 'UrlWithStringQuery & { method: Method; headers: Record<string, Primitive> & Headers; }' is not assignable to type 'RequestOptions'.
  Types of property 'headers' are incompatible.
    Type 'Record<string, Primitive> & Headers' is not assignable to type 'OutgoingHttpHeaders | undefined'.
      Type 'Record<string, Primitive> & Headers' is not assignable to type 'OutgoingHttpHeaders'.
        'string' index signatures are incompatible.
          Type 'Primitive' is not assignable to type 'OutgoingHttpHeader | undefined'.
            Type 'false' is not assignable to type 'OutgoingHttpHeader | undefined'.ts(2322)
```

This is a well known pain point of Mappersmith so I chose to type cast it because that is how we intend for it to be here, and it should be fairly safe and not dangerous to do so. It just restricts a bit what we can and cannot do in this function body.